utils/libarchive/extract/wrapper.py

```
@@ -0,0 +1,97 @@
+        if missing:
+            raise ValueError(f"Error: {missing!r} not found in archive.")
+    except Exception:
+        # Remove all files created by this call
+        for path in created_paths:
+            path.unlink(missing_ok=True)
+        raise
+
+
+def main():
+    """Main snakemake process."""
+    sys.stderr = open(snakemake.log[0], "w", buffering=1)
```

Ruff (0.15.12)

[error] Undefined name `snakemake`

(F821)